inst/include/demographic_projection.hpp

```diff
@@ -8,18 +8,16 @@ void run_demographic_projection(int time_step,
                                 const State<real_type>& state_curr,
                                 State<real_type>& state_next,
                                 WorkingData<real_type>& working) {
-  run_ageing_and_mortality(time_step, pars, state_curr, state_next, working);
+  run_ageing_and_mortality(time_step, pars, state_curr, state_next);
```

Comment:
It might be nice to keep the same signature for all steps so that you don't have to think too much about what you're passing around. We could formalise that and make a small struct that keeps all the bits together of course...

This would help later if we decided that we want to loop over some array of pointers to functions